### DISCUSSION TITLE
NC AIO Unhealthy Docker Containers after OS upgrade

### DISCUSSION THREAD

**Discussion by securegh** [nextcloud/all-in-one#1413] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Upgrading Ubuntu server to 22.04.1 LTS
2. Updating docker ce  20.10.21
3. After reboot NC AIO nextcloud and apache containers starting unhealthy
4. Stopping/Deleting all NC AIO containers and master container
5. Restarting master container and NC aio containers
```
docker run -it \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8080:8080 \
-e APACHE_PORT=11000 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest-arm64
```
7. Same issue, nextcloud and apache container starting unhealthy  

### Expected behavior 
Normal NC  AIO containers starting

### Actual behavior 
Two containers starting up unhealthy, nextcloud and apache
**nextcloud-aio-apache**

    Container logs 
    ```
    Waiting for Nextcloud to start...
    Waiting for Nextcloud to start...
    ```
**nextcloud-aio-nextcloud** 

    Container logs shows 
    ```
    Waiting for database to start...
    Waiting for database to start...
    ```

**nextcloud-aio-database** container starting normal
  Containner logs

    ```
    Setting max connections...
    PostgreSQL Database directory appears to contain a database; Skipping initialization
    2022-11-16 13:57:08.998 UTC [12] LOG:  starting PostgreSQL 14.5 on aarch64-unknown-linux-musl, compiled by gcc (Alpine 
    11.2.1_git20220219) 11.2.1 20220219, 64-bit
    2022-11-16 13:57:08.998 UTC [12] LOG:  listening on IPv4 address "0.0.0.0", port 5432
    2022-11-16 13:57:08.998 UTC [12] LOG:  listening on IPv6 address "::", port 5432
    2022-11-16 13:57:09.010 UTC [12] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
    2022-11-16 13:57:09.022 UTC [19] LOG:  database system was shut down at 2022-11-16 13:53:57 UTC
    2022-11-16 13:57:09.028 UTC [12] LOG:  database system is ready to accept connections
    ```

### Host OS 

```
Operating System: Ubuntu 22.04.1 LTS
 Kernel: Linux 5.15.0-1021-oracle
 Architecture: arm64
 Hardware Vendor: QEMU
 Hardware Model: KVM Virtual Machin
```

```
Docker version 	

Client: Docker Engine - Community
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        baeda1f
 Built:             Tue Oct 25 18:01:03 2022
 OS/Arch:           linux/arm64
 Context:           default
 Experimental:      true
```
#### Nextcloud AIO version 

Nextcloud AIO v2.1.2

#### Current channel 
latest-arm64 channel

#### Other valuable info 

This issue occurred right after the OS upgrade.   
Stopped and removed all of NC AIO and recreated them, the issue still occurs 
Switched channel to beta, still same issue. 
It seem that nextcloud container can't communicate with database container for some reason. 
```
bash-5.1# nc -zv nextcloud-aio-database 5432
nc: connect to nextcloud-aio-database port 5432 (tcp) failed: Host is unreachable

```
Here NC AIO docker network config 
[<script src="https://gist.github.com/securegh/e16c88b3f6548b3cfdff7f68cf147672.js"></script>](https://gist.github.com/securegh/e16c88b3f6548b3cfdff7f68cf147672)

Thank you for any help!

**Comment by szaimen** [comment]
Most likely is the firewall the reason why the containers dont start anymore...

**Reply by securegh** [reply]
I managed to solve this issue. 
Stopping and deleting all of NC AIO containers plus the docker **network** **nextcloud-aio**  
Stating master container and NC AIO containers.
Normal start up, Issue solved!

**Reply by securegh** [reply]
This issue looked scary at first tbh

**Comment by securegh** [comment, marked as answer]
Troubleshooting steps: 

1. Checking nextcloud container logs 
2. nextcloud waiting for database to start up
3. Testing connection between nextcloud and database container failed
4. Stopping and deleting all NC AIO containers and mastercontainer 
5. Deleting docker internal network **nextcloud-aio**
6. Starting mastercontainer 
7. Starting NC AIO containers from mastercontainer portail 
8. Normal start up, issue solved!

**Reply by szaimen** [reply]
Thanks foe sharing and grwat that you found a solution!
